x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/filestructurefinder/DelimitedFileStructureFinderFactory.java

```diff
@@ -14,6 +14,7 @@
 
 public class DelimitedFileStructureFinderFactory implements FileStructureFinderFactory {
 
+    static final double DEFAULT_BAD_ROWS_PERCENTAGE = 0.10d;
```

Comment:
I am willing to change this number. Though i do think we should have some static limit as it could break the automatic header row check if too many rows are poorly formatted.

Reply:
That's ok for me. There has to be some default and maybe in the future that setting will become user-configurable...